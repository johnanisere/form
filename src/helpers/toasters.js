import React from "react";

import SVG from "react-inlinesvg";
import { toast } from "react-toastify";
import ToastContainer from "../blocks/ToastContainer";

const config = {
  position: toast.POSITION.TOP_RIGHT
};
const BASE = "https://res.cloudinary.com/drqqv1c2v/image/upload";

const Container = ({ src, className, msg }) => (
  <ToastContainer>
    <SVG src={src} className={className} /> {msg}
  </ToastContainer>
);

export const toastError = err =>
  toast.error(
    <Container
      msg={err}
      className="toast-danger-icon"
      src={`${BASE}/v1543619703/danger.svg`}
    />,
    config
  );

export const toastSuccess = msg =>
  toast.success(
    <Container
      msg={msg}
      className="toast-success-icon"
      src={`${BASE}/v1543619743/success.svg`}
    />,
    config
  );

export const toastWarning = msg => {
  toast.warn(
    <Container
      msg={msg}
      className="toast-warning-icon"
      src={`${BASE}/v1543619814/warning.svg`}
    />,
    config
  );
};
