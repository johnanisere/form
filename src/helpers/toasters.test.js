import React from "react";
import { shallow } from "enzyme";

import * as toasters from "./toasters";

it("Toaster success component renders", () => {
  const wrapper = shallow(<toasters.toastSuccess />);
  expect(wrapper).toMatchSnapshot();
});

it("Toaster error component renders", () => {
  const wrapper = shallow(<toasters.toastError />);
  expect(wrapper).toMatchSnapshot();
});

it("Toaster warning component renders", () => {
  const wrapper = shallow(<toasters.toastWarning />);
  expect(wrapper).toMatchSnapshot();
});
