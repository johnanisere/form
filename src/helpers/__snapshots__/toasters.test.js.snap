// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster error component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <toastError />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "dpyv2dtors",
  Symbol(enzyme.__nodes__): Array [
    "dpyv2dtors",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Toaster success component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <toastSuccess />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "o3g68qhbv5",
  Symbol(enzyme.__nodes__): Array [
    "o3g68qhbv5",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Toaster warning component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <toastWarning />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
